coalib/misc/Exceptions.py

```diff
@@ -20,10 +15,10 @@ def get_exitcode(exception, log_printer=None):
         exitcode = exception.code
     elif isinstance(exception, VersionConflict):
         log_message = Constants.VERSION_CONFLICT_MESSAGE % str(exception.req)
-        log_printer.log_exception(log_message, exception)
+        LogPrinterMixin().log_exception(log_message, exception)
```

Comment:
LogPrinterMixin is not meant to be used directly AFAIU. (the log_message method returns NotImplementedError)

Edit: I just saw that you changed the log_message method in this PR. Anyway I think that the logging module can be used directly here ``logging.exception(log_message)``